src/test/java/org/folio/rest/api/CheckOutLockAPITest.java

```diff
@@ -0,0 +1,132 @@
+    checkOutLockClient.create(checkOutLock5);
+
+  }
+
+  @SneakyThrows
+  @Test
+  public void canGetCheckOutLock() {
+    String userId1 = UUID.randomUUID().toString();
+    JsonObject checkOutLock1 = toJsonObject(createCheckoutLockRequest(userId1, 1000));
+    JsonResponse response = checkOutLockClient.attemptCreate(checkOutLock1);
+    assertThat(response.getStatusCode(),is(HttpStatus.SC_CREATED));
+
+    String id = response.getJson().getString("id");
+    JsonResponse response1 = checkOutLockClient.attemptGetById(UUID.fromString(id));
+    assertThat(response1.getStatusCode(),is(HttpStatus.SC_OK));
+    assertThat(response1.getJson(),is(response.getJson()));
+
+    JsonResponse response2 = checkOutLockClient.attemptGetById("abcd");
+    assertThat(response2.getStatusCode(),is(HttpStatus.SC_BAD_REQUEST));
+    assertThat(response2.getBody(), is("Invalid lock id"));
+
+  }
+
+  @SneakyThrows
+  @Test
+  public void canGetCheckoutLocksByQueryParams() {
+    String userId1 = UUID.randomUUID().toString();
+    JsonObject checkOutLock1 = toJsonObject(createCheckoutLockRequest(userId1, 1000));
+    JsonResponse response1 = checkOutLockClient.attemptCreate(checkOutLock1);
+    assertThat(response1.getStatusCode(),is(HttpStatus.SC_CREATED));
+
+    String userId2 = UUID.randomUUID().toString();
+    JsonObject checkOutLock2 = toJsonObject(createCheckoutLockRequest(userId2, 1000));
+    JsonResponse response2 = checkOutLockClient.attemptCreate(checkOutLock2);
+    assertThat(response2.getStatusCode(),is(HttpStatus.SC_CREATED));
+
+    MultipleRecords<JsonObject> response = checkOutLockClient.getManyWithQueryParams(userId1,0,10);
+    assertThat(response.getTotalRecords(), is(1));
+
+  }
+
+  @SneakyThrows
+  @Test
+  public void canDeleteCheckOutLock() {
+    String userId1 = UUID.randomUUID().toString();
+    JsonObject checkOutLock1 = toJsonObject(createCheckoutLockRequest(userId1, 1000));
+    JsonResponse response1 = checkOutLockClient.attemptCreate(checkOutLock1);
+
+    String id = response1.getJson().getString("id");
+    checkOutLockClient.deleteById(UUID.fromString(id));
+
+    JsonObject checkOutLock2 = toJsonObject(createCheckoutLockRequest(userId1, 100000));
+    JsonResponse response2 = checkOutLockClient.attemptCreate(checkOutLock2);
+    assertThat(response2.getStatusCode(), is(HttpStatus.SC_CREATED));
+
+    TextResponse response3 = checkOutLockClient.attemptDeleteById("1234");
+    assertThat(response3.getStatusCode(), is(HttpStatus.SC_BAD_REQUEST));
+    assertThat(response3.getBody(), is("Invalid lock id"));
+  }
+
+  private CheckoutLockRequest createCheckoutLockRequest(String userId, int ttlMs) {
+    return new CheckoutLockRequest()
+      .withTtlMs(ttlMs)
+      .withUserId(userId);
+  }
+
+  private JsonObject toJsonObject(CheckoutLockRequest checkoutLockRequest)
+    throws JsonProcessingException {
+    return new JsonObject(objectMapper.writeValueAsString(checkoutLockRequest));
+  }
+
```

Comment:
Probably I'm missing something, but I don't see the most important test - to make sure it's impossible to create two locks for the same user. This is the goal of the whole feature, otherwise it wouldn't lock anything.

Reply:
I have taken care of the above mentioned scenario in the below test case.
**canCreateCheckOutLock**
Lock is created for userId1.
```
JsonObject checkOutLock1 = toJsonObject(createCheckoutLockRequest(userId1, 1000));
checkOutLockClient.create(checkOutLock1);
```
Now for the same userId, we are trying to create a lock with high ttl value. Because of high ttl, lock will not gets deleted and caller will get the error.
```
JsonObject checkOutLock2 = toJsonObject(createCheckoutLockRequest(userId1, 100000));
JsonResponse response1 = checkOutLockClient.attemptCreate(checkOutLock2);
```
Please let me know if I need to add any other scenario.

Reply:
Got it. Well, I think explicit is always better than implicit, so I personally would make it a separate test or renamed canCreateCheckOutLock to show that we're not only testing that it's possible to create a lock but also the fact that we can only create one lock per user. In any case, it's good that this case is covered.